<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>NCRDExpressionBuilderDialog</class>
 <widget class="QDialog" name="NCRDExpressionBuilderDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>814</width>
    <height>560</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Expression Builder</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2" stretch="1,10,1">
   <item>
    <layout class="QGridLayout" name="gridLayout" columnstretch="1,3,1,1,3,1">
     <property name="spacing">
      <number>4</number>
     </property>
     <item row="0" column="1">
      <widget class="NCRDItemSourceTypeCombo" name="comboSt"/>
     </item>
     <item row="1" column="1">
      <widget class="QComboBox" name="comboDataSource"/>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Field source type:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QComboBox" name="comboColumn"/>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Data source:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Column:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="2">
      <widget class="QToolButton" name="btnAddColumn">
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset resource="designer.qrc">
         <normaloff>:/designer/images/add.png</normaloff>:/designer/images/add.png</iconset>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QToolButton" name="btnAddDataSource">
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset resource="designer.qrc">
         <normaloff>:/designer/images/add.png</normaloff>:/designer/images/add.png</iconset>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Variable:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QComboBox" name="comboVar"/>
     </item>
     <item row="3" column="2">
      <widget class="QToolButton" name="btnAddVar">
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset resource="designer.qrc">
         <normaloff>:/designer/images/add.png</normaloff>:/designer/images/add.png</iconset>
       </property>
      </widget>
     </item>
     <item row="1" column="3">
      <widget class="QLabel" name="label_6">
       <property name="text">
        <string>Data source function:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="4">
      <widget class="QComboBox" name="comboDSFunction"/>
     </item>
     <item row="1" column="5">
      <widget class="QToolButton" name="btnAddDSFunc">
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset resource="designer.qrc">
         <normaloff>:/designer/images/add.png</normaloff>:/designer/images/add.png</iconset>
       </property>
      </widget>
     </item>
     <item row="2" column="3">
      <widget class="QLabel" name="label_7">
       <property name="text">
        <string>Value function:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="4">
      <widget class="QComboBox" name="comboValueFunction"/>
     </item>
     <item row="2" column="5">
      <widget class="QToolButton" name="btnAddDataFunc">
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset resource="designer.qrc">
         <normaloff>:/designer/images/add.png</normaloff>:/designer/images/add.png</iconset>
       </property>
      </widget>
     </item>
     <item row="3" column="3">
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>Parameter:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="4">
      <widget class="QComboBox" name="comboPar"/>
     </item>
     <item row="3" column="5">
      <widget class="QToolButton" name="btnAddPar">
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset resource="designer.qrc">
         <normaloff>:/designer/images/add.png</normaloff>:/designer/images/add.png</iconset>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <widget class="NCRDDataSourceTree" name="treeWidget">
      <column>
       <property name="text">
        <string notr="true">Data Sources</string>
       </property>
      </column>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QGridLayout" name="gridButtons">
         <property name="spacing">
          <number>4</number>
         </property>
         <item row="0" column="0">
          <widget class="QPushButton" name="btnPlus">
           <property name="text">
            <string>+</string>
           </property>
          </widget>
         </item>
         <item row="2" column="4">
          <widget class="QPushButton" name="btnCompareEqual">
           <property name="text">
            <string>==</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QPushButton" name="btnEqual">
           <property name="text">
            <string>=</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QPushButton" name="btnMinus">
           <property name="text">
            <string>-</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QPushButton" name="btnMulti">
           <property name="text">
            <string>*</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QPushButton" name="btnDiv">
           <property name="text">
            <string>/</string>
           </property>
          </widget>
         </item>
         <item row="1" column="4">
          <widget class="QPushButton" name="btnXor">
           <property name="text">
            <string>XOR</string>
           </property>
          </widget>
         </item>
         <item row="0" column="4">
          <widget class="QPushButton" name="btnNot">
           <property name="text">
            <string>NOT</string>
           </property>
          </widget>
         </item>
         <item row="0" column="3">
          <widget class="QPushButton" name="btnAnd">
           <property name="text">
            <string>AND</string>
           </property>
          </widget>
         </item>
         <item row="1" column="3">
          <widget class="QPushButton" name="btnOr">
           <property name="text">
            <string>OR</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QPushButton" name="btnNotEqual">
           <property name="text">
            <string>!=</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QPushButton" name="btnGreather">
           <property name="text">
            <string>&gt;</string>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QPushButton" name="btnLessEq">
           <property name="text">
            <string>&lt;=</string>
           </property>
          </widget>
         </item>
         <item row="2" column="3">
          <widget class="QPushButton" name="btnGrEq">
           <property name="text">
            <string>&gt;=</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QPushButton" name="btnLess">
           <property name="text">
            <string>&lt;</string>
           </property>
          </widget>
         </item>
         <item row="0" column="5">
          <widget class="QLabel" name="lblRole">
           <property name="text">
            <string>Role:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="6">
          <widget class="QComboBox" name="comboRole"/>
         </item>
         <item row="2" column="6">
          <widget class="QPushButton" name="btnReset">
           <property name="text">
            <string>Clear</string>
           </property>
          </widget>
         </item>
         <item row="2" column="7">
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPlainTextEdit" name="editor"/>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>NCRDItemSourceTypeCombo</class>
   <extends>QComboBox</extends>
   <header>ncrditemsourcetypecombo.h</header>
  </customwidget>
  <customwidget>
   <class>NCRDDataSourceTree</class>
   <extends>QTreeWidget</extends>
   <header>ncrddatasourcetree.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>comboSt</tabstop>
  <tabstop>comboDataSource</tabstop>
  <tabstop>comboColumn</tabstop>
  <tabstop>treeWidget</tabstop>
  <tabstop>editor</tabstop>
 </tabstops>
 <resources>
  <include location="designer.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>NCRDExpressionBuilderDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>NCRDExpressionBuilderDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
